### PR TITLE
Feature: support environment variables on Linux (bash)

### LINKED ISSUE #322: [Feature Request]: Allow environment variable creation under Linux
### Context

Labeled as "TODO" in job_environment_variables.py

### Description

Setting persistent environment variables is not supported on Linux

### Related links

https://github.com/Guts/qgis-deployment-cli/blob/16c81509da6d86737f1558dee0b74b48cccccef1/qgis_deployment_toolbelt/jobs/job_environment_variables.py#L132

### Use Cases

For linux, edit ~/.profile or add a .env file and source it from ~./profile

### Visuals

_No response_

### Before submitting

- [X] I have verified that my idea is a change request and not a bug report.

### PR DESCRIPTION
**Pull request by Guts**
- [x] add a mechanism to check if `.profile` is loaded from bashrc
- [x] allow to remove an environment variable
- [x] allow to update an environment variable (idea: look for line starting swith `export {env_value}` then check the value following the `=`)
- [x] integrate with QDT job logic and codebase

## Resources

- [Difference between ~/.profile, .bashrc and .bash_profile](https://superuser.com/a/183980)
- [.bash_profile vs .bashrc](https://joshstaiger.org/archives/2005/07/bash_profile_vs.html)

closes #322

### CODE REVIEW

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        else:
+            raise NotImplementedError(f"OS {os.name!r} support not available")
+
+        logger.debug(f"Default shell from environment variable: {shell}")
+
+    return shell
+
+
+def is_dot_profile_file() -> bool:
+    return check_path(
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:95]: Missing docstring

File: requirements/base.txt [hunk 2]
```diff
@@ -5,4 +5,5 @@ imagesize>=1.4,<1.5
 packaging>=20,<24
 pyyaml>=5.4,<7
 pywin32==306 ; sys_platform == 'win32'
+shellingham>=1.5.4,<1.6  ; sys_platform == 'linux'
```
Comment (Guts) [inline requirements/base.txt:8]: Also required for Mac?

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+    else:
+        logger.error(f"Shell {shell[0]} is not supported")
+        return False
+
+
+def get_profile_file(scope: str = "user") -> str:
+    """Get Linux profile file depending on shell and scope
+    Args:
+
+        scope (str, optional): environment variable scope. Must be "user" or "system",
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:349]:
```suggestion
    Args:
        scope (str, optional): environment variable scope. Must be "user" or "system",
```

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 4]
```diff
@@ -0,0 +1,421 @@
...
+    else:
+        logger.error(f"Shell {shell[0]} is not supported")
+        return False
+
+
+def delete_environment_variable(env_key: str, scope: str = "user") -> bool:
+    """Remove environment variable from Linux profile file
+    Args:
+
+        env_key (str): environment variable name (= key) to remove
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:244]:
```suggestion
    Args:
        env_key (str): environment variable name (= key) to remove
```

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+    return env_value
+
+
+def set_environment_variable(
+    env_key: str, env_value: str | bool | int, scope: str = "user"
+) -> bool:
+    """Set environment variable in Linux profile file
+    Args:
+
+        env_key (str): environment variable name (= key) to set
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:140]:
```suggestion
    Args:
        env_key (str): environment variable name (= key) to set
```

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 6]
```diff
@@ -0,0 +1,421 @@
...
+        must_exists=True,
+        raise_error=False,
+    )
+
+
+def get_environment_variable(envvar_name: str, scope: str = "user") -> str | None:
+    """Get environment variable from Linux profile file
+    Args:
+
+        envvar_name (str): environment variable name (= key) to retrieve
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:107]:
```suggestion
def get_environment_variable(envvar_name: str, scope: str = "user") -> str | None:
    """Get environment variable from Linux profile file
    Args:
        envvar_name (str): environment variable name (= key) to retrieve
```

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+
+# #############################################################################
+# ########## Globals ###############
+# ##################################
+
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s||%(levelname)s||%(module)s||%(lineno)d||%(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:36]: Do not override log configuration in a sub module

```suggestion
```

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py [hunk 8]
```diff
@@ -6,11 +6,12 @@
...
 """
 
-
 # #############################################################################
 # ########## Libraries #############
 # ##################################
 
+
+# Standard library
 # Standard library
```
Comment (Guts) [inline qgis_deployment_toolbelt/jobs/job_environment_variables.py:15]:
```suggestion

# Standard library
```

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py [hunk 9]
```diff
@@ -24,19 +25,42 @@
...
 # ########## Globals ###############
 # ##################################
 
+
 # logs
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="%(asctime)s||%(levelname)s||%(module)s||%(lineno)d||%(message)s",
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
```
Comment (Guts) [inline qgis_deployment_toolbelt/jobs/job_environment_variables.py:39]: Do not override log configuration a sub module

```suggestion
```

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py [hunk 10]
```diff
@@ -24,19 +25,42 @@
...
+    )
+elif opersys == "win32":
+    from qgis_deployment_toolbelt.utils.win32utils import (
+        delete_environment_variable,
+        refresh_environment,
+        set_environment_variable,
+    )
+else:
+    logger.debug("Unsupported operating system.")
+    exit()
```
Comment (Guts) [inline qgis_deployment_toolbelt/jobs/job_environment_variables.py:61]: Mac (darwin) is missing!

And the log should be critical or error at least:

```suggestion
else:
    logger.error("Unsupported operating system.")
    exit()
```

**Review by Guts — COMMENTED** [review]
A first quick review to allow async work.

File: requirements/base.txt (hunk 2 above)
Comment (sigeal) [reply requirements/base.txt:8]: I am still wondering if shellingham should be used at all, a get_profile_file() doesn't use it, but this needs to be discussed.

Comment (sigeal) [reply requirements/base.txt:8]: May we assume that linux rules to find profile file are also valid for Mac ?

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py [hunk 11]
```diff
@@ -127,7 +145,55 @@ def run(self) -> None:
...
 
-        # TODO: for linux, edit ~/.profile or add a .env file and source it from ~./profile
+        elif opersys == "linux":
+            logger.debug(f"OS : {opersys}")
+            for env_var in self.options:
+                print(
+                    f'ACTION {env_var.get("action")}',
+                    f'NAME {env_var.get("name")}',
+                    f'VALUE {env_var.get("value")}',
+                )
```
Comment (Guts) [inline qgis_deployment_toolbelt/jobs/job_environment_variables.py:155]: clean up

```suggestion
```

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py [hunk 12]
```diff
@@ -127,7 +145,55 @@ def run(self) -> None:
...
+                                value=env_var.get("value"),
+                                value_type=env_var.get("value_type"),
+                            ),
+                            scope=env_var.get("scope")
+                        )
+                    except NameError:
+                        logger.debug(
+                            f"Variable name '{env_var.get('name')}' is not defined"
+                        )
+                """
```
Comment (Guts) [inline qgis_deployment_toolbelt/jobs/job_environment_variables.py:194]: clean up?

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 13]
```diff
@@ -0,0 +1,412 @@
...
+# ##################################
+
+logger = logging.getLogger(__name__)
+
+qdt_block_comment_start = "# BEGIN QDT MANAGED BLOCK"
+qdt_block_comment_end = "# END QDT MANAGED BLOCK"
+shell_path_to_names = {
+    "bash": ("/bin/bash", "/usr/bin/bash"),
+    "zsh": ("/bin/bash/zsh", "/usr/bin/zsh"),
+}
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:39]:
```suggestion
shell_path_to_names = {
    "bash": ("/bin/bash", "/usr/bin/bash"),
    "zsh": ("/bin/bash/zsh", "/usr/bin/zsh"), # TODO: support ZSH
}
```

File: qgis_deployment_toolbelt/utils/linux_utils.py [hunk 14]
```diff
@@ -0,0 +1,412 @@
...
+            is not found.
+    """
+    for key, values in shell_path_to_names.items():
+        if value_to_find in values:
+            return key
+    return None
+
+
+def get_shell_to_use() -> tuple[str, str] | None:
+    try:
```
Comment (Guts) [inline qgis_deployment_toolbelt/utils/linux_utils.py:65]: Missing docstring

File: qgis_deployment_toolbelt/jobs/job_environment_variables.py (hunk 12 above)
Comment (sigeal) [reply qgis_deployment_toolbelt/jobs/job_environment_variables.py:189]: Done

File: qgis_deployment_toolbelt/utils/linux_utils.py (hunk 14 above)
Comment (sigeal) [reply qgis_deployment_toolbelt/utils/linux_utils.py:65]: Added

File: qgis_deployment_toolbelt/utils/linux_utils.py (hunk 1 above)
Comment (sigeal) [reply qgis_deployment_toolbelt/utils/linux_utils.py:90]: Added

**Pull request closed without merging**